### PR TITLE
Improve gemspec more

### PR DESCRIPTION
**Pull request by AlexWayfer**

### CODE REVIEW

**Comment by n-rodriguez** [comment]
Btw the next ruby release will be the 3.1, not 4. So '>= 2.4, < 4' is totally useless and could stand here for 10 years (the time to release ruby 4) without protecting you from any incompatibilities. My 2 cents.

**Comment by AlexWayfer** [comment]
> Btw the next ruby release will be the 3.1, not 4. So '>= 2.4, < 4' is totally useless and could stand here for 10 years (the time to release ruby 4) without protecting you from any incompatibilities. My 2 cents.

1. It's not related to this PR, I don't change any versions.
2. `3.1` is less than `4` (`3.1 < 4`), and it's backward compatible with `3.x`. It's [Semver](https://semver.org/).
3. It's OK to not change version (at least upper one) for ten years.

File: codecov.gemspec [hunk 1]
```diff
@@ -8,12 +8,27 @@ Gem::Specification.new do |s|
...
+  s.metadata = {
+    'source_code_uri' => github_uri,
+    'documentation_uri' =>
+      "http://www.rubydoc.info/gems/#{s.name}/#{s.version}",
+    'bug_tracker_uri' => "#{github_uri}/issues",
+    'changelog_uri' => "#{github_uri}/blob/v#{s.version}/CHANGELOG.md"
+  }
+
+  s.metadata['homepage_uri'] = github_uri
+  s.homepage = s.metadata['homepage_uri']
```
Comment (thomasrockhu) [inline codecov.gemspec:28]:
```suggestion
  s.metadata = {
    'bug_tracker_uri' => "#{github_uri}/issues",
    'changelog_uri' => "#{github_uri}/blob/v#{s.version}/CHANGELOG.md",
    'homepage_uri' => github_uri,
    'documentation_uri' =>
      "http://www.rubydoc.info/gems/#{s.name}/#{s.version}",
    'source_code_uri' => github_uri
  }
  s.homepage = github_uri
```

alphabetized and removed some indirection. I don't expect us to have a separate homepage in the near-term future.

Comment (AlexWayfer) [reply codecov.gemspec]: The point is in the thing that "homepage equals homepage" instead "this homepage equals GitHub URI and this homepage equals GitHub URI too", it's more logical and prevents multiple changes.

I've changed it a little to more beautiful way and sorted `metadata` keys alphabetically.

**Review by thomasrockhu — APPROVED** [review]
As always, thanks so much @AlexWayfer for your contributions

**Comment by AlexWayfer** [comment]
> As always, thanks so much @AlexWayfer for your contributions

Thank you for your service! ❤️ I'm glad to help you with Ruby as mush as I can.

**Pull request merged**